### ISSUE TITLE
IsValidShopDomainAsync should check if X-ShopId header has a value

### ISSUE THREAD

**Issue by waheedbhatti** [nozzlegear/ShopifySharp#995]
The current code checks if the X-ShopId header is present, but in some cases the header is present even for invalid URLs. However, even when the X-ShopId header is present there will never be a value for invalid URLs.

I suggest the following code be added to check if X-ShopId header as a value.

```
var headerValues = response.Headers.FirstOrDefault(h => h.Key.Equals("X-ShopId", StringComparison.OrdinalIgnoreCase)).Value;
if (headerValues != null && headerValues.Any(v => !string.IsNullOrWhiteSpace(v)))
{
    return true;
}
return false;
```

**Comment by nozzlegear** [comment]
Thanks for the report @waheedbhatti! I don't see any issue with the code you've provided and I'd be happy to fix the method. Just so I understand the problem clearly though, this is the code for `IsValidShopDomainAsync` right now:

```
var response = await client.SendAsync(msg, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
var hasShopIdHeader = response.Headers.Any(h => h.Key.Equals("X-ShopId", StringComparison.OrdinalIgnoreCase));

return hasShopIdHeader;
```

Are you saying this current code could incorrectly return true for invalid URLs? Is that because the invalid, non-Shopify URLs might also have their own X-ShopId header, or is there some other issue at play?

**Comment by waheedbhatti** [comment]
Yes, that's correct. The current code has incorrectly returned true for domains that are invalid. I'm not sure but what I think happens is that if a shop ever existed at the domain but no longer does, Shopify returns the X-ShopId header but with a null (empty) value.

Waheed Bhatti
________________________________

**Comment by nozzlegear** [comment]
That makes sense to me. I'm working on a release for today actually, so I'll make sure this gets fixed and published at the same time!

**Comment by waheedbhatti** [comment]
Thank you.

**Comment by nozzlegear** [comment]
Should be fixed in 6.12.0 on Nuget!
